### ISSUE TITLE
rke2 panic on startup periodically

### ISSUE THREAD

**Issue by katran001** [rancher/rke2#2716]
Issue description:
we've noticed a couple of instances where during the initial starting of the rke2 server runs into a segmentation fault
```
Mar 18 20:30:00 server2 rke2[47726]: panic: runtime error: invalid memory address or nil pointer dereference
Mar 18 20:30:00 server2 rke2[47726]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1952746]
Mar 18 20:30:00 server2 rke2[47726]: goroutine 1474 [running]:
Mar 18 20:30:00 server2 rke2[47726]: github.com/rancher/wrangler/pkg/generated/controllers/core.(*Factory).Core(0x0, 0x5, 0x6b0ecdb)
Mar 18 20:30:00 server2 rke2[47726]: /go/pkg/mod/github.com/rancher/wrangler@v0.8.11-0.20220211163748-d5a8ee98be5f/pkg/generated/controllers/core/factory.go:66 +0x26
Mar 18 20:30:00 server2 rke2[47726]: github.com/rancher/k3s/pkg/etcd.(*ETCD).Snapshot(0xc000109900, 0x7497918, 0xc0014ad880, 0xc0013f5208, 0xa489008800100001, 0x4020898000008800)
Mar 18 20:30:00 server2 rke2[47726]: /go/pkg/mod/github.com/rancher/k3s@v1.21.9-engine0.0.20220125211000-d3189ad239a5/pkg/etcd/etcd.go:979 +0x13b
Mar 18 20:30:00 server2 rke2[47726]: github.com/rancher/k3s/pkg/etcd.(*ETCD).setSnapshotFunction.func1()
Mar 18 20:30:00 server2 rke2[47726]: /go/pkg/mod/github.com/rancher/k3s@v1.21.9-engine0.0.20220125211000-d3189ad239a5/pkg/etcd/etcd.go:1603 +0x49
Mar 18 20:30:00 server2 rke2[47726]: github.com/robfig/cron/v3.FuncJob.Run(0xc000a9bf60)
Mar 18 20:30:00 server2 rke2[47726]: /go/pkg/mod/github.com/robfig/cron/v3@v3.0.1/cron.go:136 +0x25
Mar 18 20:30:00 server2 rke2[47726]: github.com/robfig/cron/v3.(*Cron).startJob.func1(0xc000f3b680, 0x73fa140, 0xc000a9bf60)
Mar 18 20:30:00 server2 rke2[47726]: /go/pkg/mod/github.com/robfig/cron/v3@v3.0.1/cron.go:312 +0x65
Mar 18 20:30:00 server2 rke2[47726]: created by github.com/robfig/cron/v3.(*Cron).startJob
Mar 18 20:30:00 server2 rke2[47726]: /go/pkg/mod/github.com/robfig/cron/v3@v3.0.1/cron.go:310 +0x73
```

RKE2 version: v1.21.10+rke2r1
installation method: offline (or airgapped, via the image tar method)

Business impact:
N/A

Troubleshooting steps:
N/A

Repro steps:
N/A

Workaround:
Is workararound available and implemented? yes/no
What is the workaround: no. curious if restarting rke2 is the play here, but customer is looking for a fix

**Comment by katran001** [comment]
Tracking internally as SURE-4297

**Comment by ShylajaDevadiga** [comment]
Reproduced the issue on v1.23.5+rke2r1 by setting the schedule to every minute at startup **etcd-snapshot-schedule-cron: "* * * * *" **
```
Apr 25 23:09:00 ip-172-31-3-164 rke2[2249]: panic: runtime error: invalid memory address or nil pointer dereference
Apr 25 23:09:00 ip-172-31-3-164 rke2[2249]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x227fa98]
Apr 25 23:09:00 ip-172-31-3-164 rke2[2249]: goroutine 514 [running]:
Apr 25 23:09:00 ip-172-31-3-164 rke2[2249]: github.com/rancher/wrangler/pkg/generated/controllers/core.(*Factory).Core(0xc000c5ac90)
Apr 25 23:09:00 ip-172-31-3-164 rke2[2249]:         /go/pkg/mod/github.com/rancher/wrangler@v0.8.11-0.20220211163748-d5a8ee98be5f/pkg/generated/controllers/core/factory.go:66 +0x18
Apr 25 23:09:00 ip-172-31-3-164 rke2[2249]: github.com/k3s-io/k3s/pkg/etcd.(*ETCD).Snapshot(0xc001ae8940, {0x70eb4b8, 0xc000a29140}, 0xc000921b08)
Apr 25 23:09:00 ip-172-31-3-164 rke2[2249]:         /go/pkg/mod/github.com/k3s-io/k3s@v1.23.5-rc4.0.20220330143550-313aaca547f0/pkg/etcd/etcd.go:1238 +0xd2
Apr 25 23:09:00 ip-172-31-3-164 rke2[2249]: github.com/k3s-io/k3s/pkg/etcd.(*ETCD).setSnapshotFunction.func1()
Apr 25 23:09:00 ip-172-31-3-164 rke2[2249]:         /go/pkg/mod/github.com/k3s-io/k3s@v1.23.5-rc4.0.20220330143550-313aaca547f0/pkg/etcd/etcd.go:1863 +0x29
Apr 25 23:09:00 ip-172-31-3-164 rke2[2249]: github.com/robfig/cron/v3.FuncJob.Run(0x6814098)
Apr 25 23:09:00 ip-172-31-3-164 rke2[2249]:         /go/pkg/mod/github.com/robfig/cron/v3@v3.0.1/cron.go:136 +0x1a
Apr 25 23:09:00 ip-172-31-3-164 rke2[2249]: github.com/robfig/cron/v3.(*Cron).startJob.func1()
Apr 25 23:09:00 ip-172-31-3-164 rke2[2249]:         /go/pkg/mod/github.com/robfig/cron/v3@v3.0.1/cron.go:312 +0x6a
Apr 25 23:09:00 ip-172-31-3-164 rke2[2249]: created by github.com/robfig/cron/v3.(*Cron).startJob
Apr 25 23:09:00 ip-172-31-3-164 rke2[2249]:         /go/pkg/mod/github.com/robfig/cron/v3@v3.0.1/cron.go:310 +0xb2
```

Validated the fix on rke2 version v1.23.6-rc3+rke2r1 with same setting on multiple installs, no panic occurred
